app/scripts/lib/crypto/recovery-keys.js

```diff
@@ -0,0 +1,175 @@
+     *
+     * https://github.com/mozilla/fxa-auth-server/blob/master/docs/recovery_keys.md
+     *
+     * @param {String} recoveryKeyId Recovery key id used in encryption
+     * @param {String} recoveryEnc Recovery encryption used to encrypt bundle
+     * @param {Object} recoveryData Data to encrypt
+     * @param {Object} options
+     * @returns {Promise} A promise that will be fulfilled with the encrypted recoveryData
+     */
+    createRecoveryBundle: function (recoveryKeyId, recoveryEnc, recoveryData, options = {}) {
+      return importFxaCryptoDeriver().then((fxaCryptoDeriver) => {
+        const jose = fxaCryptoDeriver.jose;
+
+        if (! recoveryKeyId) {
+          throw new Error('Missing recoveryKeyId');
+        }
+
+        if (! recoveryEnc) {
+          throw new Error('Missing recoveryEnc');
+        }
+
+        const keyOptions = {
+          alg: 'A256GCM',
+          k: jose.util.base64url.encode(recoveryEnc, 'hex'),
+          kty: 'oct'
+        };
+
+        return jose.JWK.asKey(keyOptions)
+          .then(function (key) {
+            const recipient = {
+              header: {
+                alg: 'dir',
+                enc: 'A256GCM',
+                kid: recoveryKeyId
+              },
+              key
+            };
+            const encryptOptions = {
+              contentAlg: 'A256GCM',
+              format: 'compact'
+            };
+
+            if (options.iv) {
+              encryptOptions.iv = options.iv;
+            }
+
+            return jose.JWE.createEncrypt(encryptOptions, recipient)
+              .update(JSON.stringify(recoveryData))
+              .final();
+          });
+      });
+    },
+
+    /**
+     * Unbundle the encoded recovery data using the specified recovery key.
+     *
+     * @param {String} recoveryEnc Recovery encryption key
+     * @param {String} encryptedRecoveryData Base64 encoded and encrypted recovery data
+     * @returns {Promise} A promise that will be fulfilled with the decoded recoveryData
+     */
+    unbundleRecoveryData: function (recoveryEnc, encryptedRecoveryData) {
+      return importFxaCryptoDeriver().then((fxaCryptoDeriver) => {
+        const jose = fxaCryptoDeriver.jose;
+
+        if (! recoveryEnc) {
+          throw new Error('Missing recoveryEnc');
+        }
+
+        if (! encryptedRecoveryData) {
+          throw new Error('Missing encryptedRecoveryData');
+        }
+
+        const keyOptions = {
+          alg: 'A256GCM',
+          k: jose.util.base64url.encode(recoveryEnc, 'hex'),
+          kty: 'oct'
+        };
+
+        return jose.JWK.asKey(keyOptions)
+          .then(function (key) {
+            const opts = {
+              algorithms: ['dir', 'A256GCM']
+            };
+
+            return jose.JWE.createDecrypt(key, opts)
+              .decrypt(encryptedRecoveryData)
+              .then(function (result) {
+                return JSON.parse(result.plaintext.toString());
```

Comment:
What is `result.plaintext` if it's not already a string?

Reply:
> What is result.plaintext if it's not already a string?

This is a byte array (specifically a node `Buffer` instance).